### PR TITLE
58742_follow_unfollow_user

### PR DESCRIPTION
**Pull request by ahammad-reyad**
Implemented follow unfollow user

### CODE REVIEW

File: app/controllers/relationships_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class RelationshipsController < ApplicationController
+  before_action :logged_in_user
+
+  def create
+    user = User.find(params[:followed_id])
```
Comment (MuhammadTamzid) [inline app/controllers/relationships_controller.rb]: remove ()

File: app/controllers/relationships_controller.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    current_user.follow user
+    redirect_to user
+  end
+
+  def destroy
+    user = Relationship.find(params[:id]).followed
```
Comment (MuhammadTamzid) [inline app/controllers/relationships_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 3]
```diff
@@ -36,6 +38,18 @@ def update
     end
   end
 
+  def following
```
Comment (MuhammadTamzid) [inline app/controllers/users_controller.rb]: RESTFul

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -36,6 +38,18 @@ def update
...
+    @title = t :following
+    @users = @user.following.paginate(page: params[:page])
+    render :show_follow
+  end
+
+  def followers
```
Comment (MuhammadTamzid) [inline app/controllers/users_controller.rb]: RESTFul

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -36,6 +38,18 @@ def update
     end
   end
 
+  def following
+    @title = t :following
+    @users = @user.following.paginate(page: params[:page])
```
Comment (MuhammadTamzid) [inline app/controllers/users_controller.rb]: remove ()

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -36,6 +38,18 @@ def update
...
+    render :show_follow
+  end
+
+  def followers
+    @title = t :followers
+    @users = @user.followers.paginate(page: params[:page])
```
Comment (MuhammadTamzid) [inline app/controllers/users_controller.rb]: remove ()

File: app/models/user.rb [hunk 7]
```diff
@@ -43,4 +43,16 @@ def forget
     update_attributes remember_digest: nil
   end
 
+  def follow other_user
+    active_relationships.create(followed_id: other_user.id)
```
Comment (MuhammadTamzid) [inline app/models/user.rb]: remove ()

File: app/views/users/_unfollow.html.erb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+<%= form_for current_user.active_relationships.find_by(followed_id: @user.id),
+  html: { method: :delete } do |f| %>
```
Comment (MuhammadTamzid) [inline app/views/users/_unfollow.html.erb]: ``
{method: :delete}
``

File: config/routes.rb [hunk 9]
```diff
@@ -6,5 +6,10 @@
   get    "login"  => "sessions#new"
   post   "login"  => "sessions#create"
   delete "logout" => "sessions#destroy"
-  resources :users
+  resources :users do
+    member do
```
Comment (dieunb) [inline config/routes.rb]: can remove this

File: app/controllers/relationships_controller.rb [hunk 10]
```diff
@@ -0,0 +1,27 @@
+class RelationshipsController < ApplicationController
+  before_action :logged_in_user
+
+  def index
+    user = User.find params[:user]
```
Comment (dieunb) [inline app/controllers/relationships_controller.rb]: i can refactor this method is below:
```
user = User.find params[:user]

@title = t params[:type].to_sym
@users = user.send(params[:type]).paginate(page: params[:page])
```

Comment (ahammad-reyad) [inline app/controllers/relationships_controller.rb]: Improved

**Pull request merged**